wvd-templates/DSC/Functions.ps1

```diff
@@ -222,13 +222,159 @@ function ExtractDeploymentAgentZipFile {
+    param(
+        [string]$Source = 'attached',
+        [string]$ArtifactsPath
+
+        # [Parameter(mandatory = $false)]
+        # [string]$DeployAgentLocation = 'C:\DeployAgent'
+    )
+
+    #//todo move outside
+    # Write-Log -Message "Creating a folder inside rdsh vm for extracting deployagent zip file"
+    # ExtractDeploymentAgentZipFile -ScriptPath $ScriptPath -DeployAgentLocation $DeployAgentLocation
+    
+    # Write-Log -Message "Changing current folder to Deployagent folder: $DeployAgentLocation"
+    # Set-Location "$DeployAgentLocation"
+    
+    if ($Source -eq 'attached') {
+        if ((-not $ArtifactsPath) -or (-not (test-path $ArtifactsPath))) {
+            throw "invalid param: ArtifactsPath = '$ArtifactsPath'"
+        }
+
+        # Locating and extracting PowerShellModules.zip
+        $ZipName = 'PowerShellModules.zip'
+        Write-Log -Message "Locating '$ZipName' within Custom Script Extension folder structure: '$ArtifactsPath'"
+        $ZipPath = (Get-ChildItem "$ArtifactsPath\" -Filter $ZipName -Recurse | Select-Object -First 1).FullName
+        if ((-not $ZipPath) -or (-not (Test-Path $ZipPath))) {
+            throw "'$ZipName' file not found at '$ArtifactsPath'"
+        }
+        
+        $Path = 'C:\PowerShellModules\'
+        if (test-path $Path) {
+            Remove-Item -Path $Path -Force -Recurse -ErrorAction 'SilentlyContinue'
+        }
+
+        Write-Log -Message "Extracting '$ZipPath' file into '$Path' folder inside VM"
+        Expand-Archive $ZipPath -DestinationPath $Path -Force -ErrorAction 'SilentlyContinue'
+
+        $DLLName = 'Microsoft.RDInfra.RDPowershell.dll'
+        Write-Log -Message "Locating '$DLLName' within '$Path'"
+        $DLLPath = (Get-ChildItem "$Path\" -Filter $DLLName -Recurse | Select-Object -First 1).FullName
+        if ((-not $DLLPath) -or (-not (Test-Path $DLLPath))) {
+            throw "'$DLLName' file not found at '$Path'"
+        }
+
+        Write-Log -Message "Importing WVD PowerShell modules from attached artifacts"
+        Import-Module $DLLPath
+        Write-Log -Message "Imported Windows Virtual Desktop PowerShell modules successfully from attached artifacts"
+        return
+    }
+
+    $Version = ($Source.Trim().ToLower() -split 'gallery@')[1]
+    if ($Version -eq $null -or $Version.Trim() -eq '') {
+        throw "invalid param: Source = $Source"
+    }
+
+    Write-Log -Message "Installing & importing WVD PowerShell modules (version: v$Version) from PowerShell Gallery"
+    if ($Version -eq 'latest') {
+        Install-Module -Name Microsoft.RDInfra.RDPowershell -Scope 'CurrentUser' -AllowClobber -Force
+    }
+    else {
+        Install-Module -Name Microsoft.RDInfra.RDPowershell -Scope 'CurrentUser' -AllowClobber -Force -RequiredVersion (new-object System.Version($Version))
+    }
+    Write-Log -Message "Installed & imported Windows Virtual Desktop PowerShell modules (version: v$Version) successfully from PowerShell Gallery"

```

Comment:
The Install-Module command doesn't import the module. I'm guessing this still works because when a command from the RDPowershell  module is run, such as  Add-RdsAccount , it will probably automatically load the module. But I think we might as well run Import-Module  here since this code already does that when it loads RDPowershell   from the local dll.

Reply:
Good idea. I'll add import-module